src/app/qgisapp.cpp

```diff
@@ -7643,6 +7643,20 @@ bool QgisApp::openLayer( const QString &fileName, bool allowInteractive )
     ok = static_cast< bool >( addMeshLayerPrivate( fileName, fileInfo.completeBaseName(), QStringLiteral( "mdal" ), false ) );
   }
 
+  // maybe a known file type, which couldn't be opened due to a missing dependency... (eg. las for a non-pdal-enabled build)
+  {
```

Comment:
Just wondering, shouldn't be this blok within `if ( !ok )` ?

Reply:
Good catch! The las/laz files return earlier then this, which is why I didn't notice anything in testing. Will fix!

Reply:
Fixed in https://github.com/qgis/QGIS/pull/41958